src/main/java/parser/AuthorArgumentFinder.java

```diff
@@ -0,0 +1,31 @@
+package parser;
+
+import exceptions.NoAuthorProvidedException;
+import exceptions.TantouException;
+
+import static constants.Options.AUTHOR_OPTION;
+import static constants.Regex.AUTHOR_NAME_EXTRACT0R_PATTERN;
+
+public class AuthorArgumentFinder extends ArgumentFinder {
+    public AuthorArgumentFinder() {
+        super(AUTHOR_NAME_EXTRACT0R_PATTERN, AUTHOR_OPTION);
+    }
+
+    @Override
+    public ArgumentResult getArgumentResult(String userInput) throws TantouException {
+        String authorName = null;
+        matcher = pattern.matcher(userInput);
+
+        if (matcher.find()) {
+            authorName = matcher.group(0).trim();
+            if (authorName.isEmpty()) {
+                throw new NoAuthorProvidedException();
+            }
+
+            // Remove the argument and its matching flag
+            userInput = userInput.substring(0, matcher.start(0) - 3) + userInput.substring(matcher.end(0));
```

Comment:
Perhaps a more descriptive variable name would be better than reassigning it?

Reply:
Referring to `userInput` for this comment and all other comments referencing this one.

Reply:
Thank you for pointing this out. Will make the required changes.